src/vs/platform/extensionManagement/node/extensionManagementService.ts

```diff
@@ -372,27 +372,31 @@ export class ExtensionManagementService extends AbstractExtensionManagementServi
 
 	private async collectFiles(extension: ILocalExtension): Promise<IFile[]> {
 
-		const collectFilesFromDirectory = async (dir: string): Promise<string[]> => {
+		const collectFilesFromDirectory = async (dir: string): Promise<IFile[]> => {
 			let entries = await pfs.Promises.readdir(dir);
 			entries = entries.map(e => path.join(dir, e));
 			const stats = await Promise.all(entries.map(e => fs.promises.stat(e)));
-			let promise: Promise<string[]> = Promise.resolve([]);
+			let promise: Promise<IFile[]> = Promise.resolve([]);
 			stats.forEach((stat, index) => {
-				const entry = entries[index];
+				const localPath = entries[index];
 				if (stat.isFile()) {
-					promise = promise.then(result => ([...result, entry]));
+					promise = promise.then(result => ([...result, {
+						path: `extension/${path.relative(extension.location.fsPath, localPath)}`,
+						localPath,
+						// chmod u+w,g+w
+						mode: stat.mode | 0o220,
```

Comment:
u+w (0o200) should be enough here, not sure if group permissions matter, added just in case.
It is also possible to change mode during extension unpack phase, but I didn't want to alter behavior for users whose extension directory is not immutable.